docs/swagger.yaml

```diff
@@ -0,0 +1,381 @@
+        name: "user_id"
+        description: "The user that needs to be fetched. "
+        required: true
+        type: 'integer'
+      operationId: "getTasks"
+      produces:
+      - "application/json"
+      responses:
+        "200":
+          description: "successful operation"
+          schema:
+            type: "array"
+            items:
+              $ref: "#/definitions/Task"
+        "500":
+          description: "Internal server error"
+    post:
+      tags:
+      - "/users/{user_id}/tasks"
+      summary: "Add a new task for a specific user to the database"
+      description: ""
+      operationId: "addTask"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
+      parameters:
+      - in: "path"
+        name: "user_id"
+        description: "The user that needs to be fetched. Use user1 for testing. "
+        required: true
+        type: 'integer'
+      - in: "body"
+        name: "body"
+        description: "Task object that needs to be added to the store"
+        required: true
+        schema:
+          $ref: "#/definitions/Task"
+      responses:
+        "200":
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/Task"
+        "404":
+          description: "User not found"
+        "500":
+          description: "Internal server error"
+    
+  /users/{user_id}/tasks/{task_id}:
+    get:
+      tags:
+      - "users/{user_id}/tasks/{task_id}"
+      summary: "Get a user's task by task_id"
+      description: ""
+      operationId: "getTaskByName"
+      produces:
+      - "application/json"
+      parameters:
+      - name: "user_id"
+        in: "path"
+        description: "The id of the user who owns the task."
+        required: true
+        type: "integer"
+      - name: "task_id"
+        in: "path"
+        description: "The task id that needs to be fetched."
+        required: true
+        type: "integer"
+      responses:
+        "200":
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/User"
+        "400":
+          description: "Invalid username supplied"
+        "404":
+          description: "User not found"
+    put:
+      tags:
+      - "users/{user_id}/tasks/{task_id}"
+      summary: "Update a user's task with PUT method"
+      description: "This can only be done by the logged in user."
+      operationId: "updateTask"
+      produces:
+      - "application/json"
+      parameters:
+      - name: "user_id"
+        in: "path"
+        description: "The id of the user who owns the task."
+        required: true
+        type: "string"
+      - name: "task_id"
+        in: "path"
+        description: "The task id that will be updated."
+        required: true
+        type: "integer"
+      - in: "body"
+        name: "body"
+        description: "Updated user object"
+        required: true
+        schema:
+          $ref: "#/definitions/User"
```

Comment:
requires a Task object, with title, description, dueDate (dd/mm/yy hh:mm:ss), completed ("0" or "1"), and completedDate (same)
I'm going to file a bug because I don't think all 5 fields should be required for a PUT, but this is how it is at the moment

Reply:
Actually the PATCH is more lenient, it take any field I think